Create Gift Card page

phillipet opened the pull request:
- Grid gift card page
  + Show/Hide editing column etGrid 
  + Add more action column optional etGrid
  + Fetch, delete record
- Modal QRCode
   + show/hide each image code of row
   + add watch event for GenQRCode module ( re-draw QCode image)
- Modal create gift card
  + Create gift card
  + Add validator

tri-et committed:
server side code gift card

tri-et committed:
changes requested

Se7en-et committed:
Merge remote-tracking branch 'tri-et/feature/server-side-code-gift-card' into feature/giftcards-page

Se7en-et committed:
Merge remote-tracking branch 'hemvanh/develop' into feature/giftcards-page

Se7en-et committed:
Create grid gift card

phillipet committed:
create modal gift card

phillipet committed:
Merge branch 'feature/giftcards-page' into feature/create-gift-card

phillipet committed:
apply responsive design layout for modal create gift card 

inherit ETModal, need implement some methods, actions

phillipet committed:
apply function show hide modal create giftcard modal

phillipet committed:
Merge branch 'develop' into feature/create-gift-card

phillipet committed:
Gen gift card code successfully, but don't able bind to text filed

Se7en-et committed:
Create gift card page

Se7en-et committed:
Merge remote-tracking branch 'hemvanh/develop' into feature/giftcards-page

phillipet committed:
bind gen gift card to textfiled

Se7en-et committed:
Merge remote-tracking branch 'hemvanh/develop' into feature/giftcards-page

Se7en-et committed:
Create delete

phillipet committed:
gen code and view qrcode image

phillipet committed:
remove gen code feature

phillipet committed:
finetune inherit et-modal component

Se7en-et committed:
fix create grid gift card and write changelog

phillipet committed:
add validator modal create gift card

phillipet committed:
remove inherit etModal

phillipet committed:
Merge branch 'feature/giftcards-page' into feature/create-gift-card

merge grid from seven

phillipet committed:
Merge branch 'develop' into create-giftcard-page

phillipet committed:
fix msg validator

phillipet committed:
load grid and create gift card success, but append to grid record failed

phillipet committed:
add new record to grid, add QRCode Modal

phillipet committed:
Merge branch 'develop' into create-giftcard-page

phillipet committed:
add qrcode attribute for QRCodeModal to pass gencode from parent grid

phillipet committed:
add more custom column cell in etGrid and action popup qrcode

phillipet committed:
delegate function and transfer param giftcardGrid & etGrid together

phillipet committed:
add watch GenQRCode comp, Re-draw image after "qrcode" attribute changed

phillipet committed:
clean code and add loading icon save button

phillipet committed:
Change log

phillipet committed:
remove slot etModal

hemvanh commented on src/components/CreateGiftCardModal.vue:
```diff
@@ -0,0 +1,81 @@
+  <q-modal v-model="isModalOpened" no-backdrop-dismiss no-esc-dismiss>
+    <q-modal-layout>
+      <q-toolbar color="primary">
+        <q-btn class="modal-title" flat color="white" icon="close" @click="isModalOpened=false"></q-btn>
```
color="white" -> replace with var ?

hemvanh commented on src/components/CreateGiftCardModal.vue:
```diff
@@ -0,0 +1,81 @@
+            <q-input v-model="expiry" type="number" suffix="days" color="secondary" :error="$v.expiry.$error" />
+            <et-validator :dirty="$v.expiry.$dirty" :show="!$v.expiry.required" msg="expiry is required" />
+            <et-validator :dirty="$v.expiry.$dirty" :show="!$v.expiry.integer" msg="expiry must have a interger" />
+            <et-validator :dirty="$v.expiry.$dirty" :show="!$v.expiry.between" msg="expiry is <=1 and <= 100 " />
```
<= 1000 ?

hemvanh commented on src/components/CreateGiftCardModal.vue:
```diff
@@ -0,0 +1,81 @@
+  </q-modal>
+</template>
+<script>
+import etModal from './EtModal'
```
etModal  is used for ?

hemvanh commented on src/components/EtModal.vue:
```diff
@@ -15,6 +15,7 @@
         <q-field class="q-mb-md" :key="field.name" v-for="field in getFields" v-if="!field.hidden" :label-width="3" :icon="field.icon" :label="field.label" :helper="field.desc" error-label="Some error">
           <q-input v-model="getEditingRec[field.name]" :type="field.type" color="secondary" />
         </q-field>
+        <slot></slot>
```
already remove slot ?

hemvanh commented on src/components/EtGrid.vue:
```diff
@@ -47,6 +51,27 @@ export default {
+      default: false,
+      type: Boolean,
+    },
+    disableExpandCol: {
```
disableExpandCol : this prop is used for ?

hemvanh commented on src/components/EtGrid.vue:
```diff
@@ -47,6 +51,27 @@ export default {
+      default: true,
+      type: Boolean,
+    },
+    expandedCol: {
```
this is a general ETModal class, what if i want to add 2 more extra action buttons ?
so expandedCol should be expandedCols, which is an array of Objects

phillipet committed:
remove prop un-use etGird

hemvanh commented on src/components/GenQRCode.vue:
```diff
@@ -37,6 +37,20 @@ export default {
     })
   },
+  watch: {
+    qrcode(newQRCode, oldQRCode) {
```
should put this into a computed so in case then qrcode change, the QR Code image change, no more duplicated code

phillipet committed:
remove some codes

phillipet committed:
fix qrcode image center popup

phillipet committed:
fix name function, remove class modal-title of modal

phillipet committed:
create method drawImage GenQRCode

phillipet committed:
format date time createdDateAt (mm/dd/yy hh:mm:ss)

format: val => `${val}%` src : https://quasar-framework.org/components/datatable.html#Alternative-custom-rows
